lib/functions/configuration/interactive.sh

```diff
@@ -151,7 +151,7 @@ function interactive_config_ask_branch() {
 }
 
 function interactive_config_ask_release() {
-	if [[ $KERNEL_ONLY != yes && -z $RELEASE ]]; then
+	if [[ -z "$BUILD_ONLY" || "$BUILD_ONLY" == *bootstrap* && -z "$RELEASE" ]]; then
```

Comment:
Give `BUILD_ONLY` a default value? So we don't need this confused condition which may have some operation priority problem.

Reply:
Currently `interactive_config_ask_release()` is never reached anymore - @The-going did comment the call in `interactive_config_ask_kernel()`.

Reply:
@hzyitc I totally agree. Initially I had this:
```diff
Subject: [PATCH 3/6] The RELEASE variable cannot be empty.

Additional conditions for checking RELEASE are unacceptable.
---
 lib/functions/configuration/interactive.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/functions/configuration/interactive.sh b/lib/functions/configuration/interactive.sh
index 9865b364e..b2f667441 100644
--- a/lib/functions/configuration/interactive.sh
+++ b/lib/functions/configuration/interactive.sh
@@ -151,7 +151,7 @@ function interactive_config_ask_branch() {
 }
 
 function interactive_config_ask_release() {
-	if [[ -z "$BUILD_ONLY" && -z "$RELEASE" ]]; then
+	if [[ -z "$RELEASE" ]]; then
 
 		options=()
 
```
This is very similar to the undocumented things mentioned above by Igor.
We'll have to fix it right away:

> KERNEL_ONLY in [CI scripts](https://github.com/armbian/scripts/tree/master/.github/workflows) is what worries me

Reply:
Let's make it clean right away.
```
if [[ -z "$RELEASE" ]]; then
```
If scripts .github/workflows will issue an error and we will fix them.

Reply:
@hzyitc Do you allow this?